server/app/transcribe.py

```diff
@@ -106,12 +107,18 @@ def process_audio(
 
     else:
         task.state = TranscriptionState.DIARIZING
-        diarization_model = BinaryKeyDiarizationModel()
-        diarization_model.CLUSTERING_SELECTION_MAX_SPEAKERS = diarize_max_speakers
-        segments = diarization_model.diarize(
-            SAMPLE_RATE, np.array(audio.get_array_of_samples())
-        )
-        optimized_segments = optimize_segments(segments)
+        try:
+            diarization_model = BinaryKeyDiarizationModel()
+            diarization_model.CLUSTERING_SELECTION_MAX_SPEAKERS = diarize_max_speakers
+            segments = diarization_model.diarize(
+                SAMPLE_RATE, np.array(audio.get_array_of_samples())
+            )
+            optimized_segments = optimize_segments(segments)
+        except:  # noqa: E722
+            traceback.print_exc()
+            optimized_segments = [
```

Comment:
we probably want to display a warning if diarization failed. though we can do this in another pr

Reply:
I opened #273 for that